department-of-veterans-affairs/caseflow
# Lianna/appeals 7431: Creation of Unassigned tab in the CSP organization queue

Resolves [APPEALS-7431](https://vajira.max.gov/browse/APPEALS-7431)

### Description
Created new **unassigned** tab to display cases in the Vha Caregiver Support Program organization, that have a VhaDocumentSearchTask status of Assigned. This solves the current user story request:  As a CSP user, I need to access Caregiver cases from the Unassigned tab so that I can begin processing document requests.

### Acceptance Criteria
- [x] Creation of the Unassigned tab in the CSP organization queue
- [x] The following informational text appears under the Unassigned tab
  - [x] Cases assigned to VHA Caregiver Support Program:
- [x] The Unassigned tab contains the following columns
  - [x] Case Details
  - [x] Assigned By
  - [x] Docket
  - [x] Days Waiting
  - [x] Veteran Documents

### Testing Plan
1. Go to Caseflow App in browser
2. Switch user to a Caregiver User
3. Select 'Your Queue' in the App Selector Queue Tab
<img width="500" alt="Screen Shot 2022-08-04 at 5 35 53 PM" src="https://user-images.githubusercontent.com/17394371/182973477-74c2b51b-553a-4228-a413-7a4385660a03.png">
4. In the Switch Views dropdown select 'VHA Caregiver Support Program team cases'
<img width="1000" alt="Screen Shot 2022-08-04 at 5 38 17 PM" src="https://user-images.githubusercontent.com/17394371/182973755-e8159d45-040d-4dba-b86d-fe78d3dcfd38.png">

5. Only visible tab will be the Unassigned Task Tab
  a. Following Text Should be visible "Cases assigned to VHA Caregiver Support Program:"
  b. Following Columns Should be visible: 
    i. Case Details
    ii. Tasks
    iiI. Assigned By
    iv. Types
    v. Docket
    vi. Days Waiting
    ii. Veteran Documents
<img width="1447" alt="Screen Shot 2022-08-04 at 5 40 50 PM" src="https://user-images.githubusercontent.com/17394371/182974117-487c3507-ecdc-4d5c-b195-0552954da101.png">
6. In Terminal open Rails console: `make c`

7. Create an appeal with a Pre-Docket Task `appeal = FactoryBot.create(:appeal, :with_pre_docket_task)`

8. Check the task tree to make sure the appeal was correctly created: `appeal.treee`
<img width="1377" alt="Screen Shot 2022-08-04 at 6 32 30 PM" src="https://user-images.githubusercontent.com/17394371/182974568-a0df5ca7-cb8e-4b49-b111-82f3ac2b55b1.png">
9. Define task by Task ID (note the ID _7128_ is from the example): `task=Task.find(7128)`

10. Create the Vha Document Search Task:
      `vha_document_search_task=VhaDocumentSearchTask.create!( parent: task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: VhaCaregiverSupport.singleton )`

11. Confirm in task tree that Vha Document Search Task has been created: `appeal.treee`
<img width="1511" alt="Screen Shot 2022-08-04 at 6 34 54 PM" src="https://user-images.githubusercontent.com/17394371/182974809-7756c39c-a935-45a0-8a74-ae436e7e5665.png">

12. Return to Caseflow in the browser and refresh page. The created appeal should be visible:
<img width="1433" alt="Screen Shot 2022-08-04 at 6 35 41 PM" src="https://user-images.githubusercontent.com/17394371/182974900-e15b7b27-800c-471e-838f-15c753fabd4f.png">

13. Insure that appeals **without** "assigned" statuses are not visible in this tab by completing the following steps:
   a. Return to rails console and change the VhaDocumentSearchTask Status to "completed":
      `Task.find(<VhaDocumentSearchTask_ID>).completed!`
   b. Check task tree to confirm status changed: `appeal.treee`
   c. Return to caseflow in the browser and refresh the page, the original appeal should no longer appear in the list since its status changed

14. Repeat Step 13 for other statuses:
    - [x] `Task.find(<VhaDocumentSearchTask_ID>).on_hold!`
    - [x] `Task.find(<VhaDocumentSearchTask_ID>).cancelled!`
    - [x] `Task.find(<VhaDocumentSearchTask_ID>).in_progress!`
